firmware/src/core/utils/MainHelper.cpp

```diff
@@ -0,0 +1,226 @@
+WiFiManager *MainHelper::s_wifiManager = nullptr;
+ConfigManager *MainHelper::s_configManager = nullptr;
+ScreenManager *MainHelper::s_screenManager = nullptr;
+WidgetSet *MainHelper::s_widgetSet = nullptr;
+int MainHelper::s_widgetCycleDelay = WIDGET_CYCLE_DELAY;
+unsigned long MainHelper::s_widgetCycleDelayPrev = 0;
+bool MainHelper::s_invertedOrbs = INVERTED_ORBS;
+std::string MainHelper::s_timezoneLocation = TIMEZONE_API_LOCATION;
+std::string MainHelper::s_ntpServer = NTP_SERVER;
+int MainHelper::s_tftBrightness = TFT_BRIGHTNESS;
+bool MainHelper::s_nightMode = DIM_ENABLED;
+int MainHelper::s_dimStartHour = DIM_START_HOUR;
+int MainHelper::s_dimEndHour = DIM_END_HOUR;
+int MainHelper::s_dimBrightness = DIM_BRIGHTNESS;
+
+void MainHelper::init(WiFiManager *wm, ConfigManager *cm, ScreenManager *sm, WidgetSet *ws) {
+    s_wifiManager = wm;
+    s_configManager = cm;
+    s_screenManager = sm;
+    s_widgetSet = ws;
+}
+
+/**
+ * The ISR handlers must be static
+ */
+void MainHelper::isrButtonChangeLeft() { buttonLeft.isrButtonChange(); }
+void MainHelper::isrButtonChangeMiddle() { buttonOK.isrButtonChange(); }
+void MainHelper::isrButtonChangeRight() { buttonRight.isrButtonChange(); }
+
+void MainHelper::setupButtons() {
+    buttonLeft.begin();
+    buttonOK.begin();
+    buttonRight.begin();
+
+    attachInterrupt(digitalPinToInterrupt(BUTTON_LEFT), isrButtonChangeLeft, CHANGE);
+    attachInterrupt(digitalPinToInterrupt(BUTTON_OK), isrButtonChangeMiddle, CHANGE);
+    attachInterrupt(digitalPinToInterrupt(BUTTON_RIGHT), isrButtonChangeRight, CHANGE);
+}
+
+void MainHelper::setupConfig() {
+    s_configManager->addConfigString("General", "timezoneLoc", &s_timezoneLocation, 30, "Timezone Location, use one from <a href='https://timezonedb.com/time-zones' target='blank'>this list</a>");
+    s_configManager->addConfigInt("General", "widgetCycDelay", &s_widgetCycleDelay, "Automatically cycle widgets every X seconds, set to 0 to disable");
+    s_configManager->addConfigString("General", "ntpServer", &s_ntpServer, 30, "NTP server", true);
+
+    s_configManager->addConfigBool("TFT Settings", "invertedOrbs", &s_invertedOrbs, "Inverted Orbs (enable if using InfoOrbs upside down)");
+    s_configManager->addConfigBool("TFT Settings", "nightmode", &s_nightMode, "Enable Nighttime mode");
+    s_configManager->addConfigInt("TFT Settings", "tftBrightness", &s_tftBrightness, "TFT Brightness [0-255]", true);
+    String optHours[] = {"0:00", "1:00", "2:00", "3:00", "4:00", "5:00", "6:00", "7:00", "8:00", "9:00", "10:00", "11:00",
+                         "12:00", "13:00", "14:00", "15:00", "16:00", "17:00", "18:00", "19:00", "20:00", "21:00", "22:00", "23:00"};
+    s_configManager->addConfigComboBox("TFT Settings", "dimStartHour", &s_dimStartHour, optHours, 24, "Nighttime Start [24h format]", true);
+    s_configManager->addConfigComboBox("TFT Settings", "dimEndHour", &s_dimEndHour, optHours, 24, "Nighttime End [24h format]", true);
```

Comment:
Wouldn't a `configInt` be easier here?
Just 0..23 ?

Reply:
Yes, int would be easier. But this looks nicer in the WebPortal.
And 0..23 should be enough, at least on earth, right? :-)